server/cluster/cluster.go

```diff
@@ -1781,3 +1784,42 @@ func newCacheCluster(c *RaftCluster) *cacheCluster {
 		stores:      c.GetStores(),
 	}
 }
+
+// Evict the leaders when the store is damaged. Damaged regions are emergency errors
+// and requires user to manually remove the `evict-leader-scheduler` with pd-ctl
+func (c *RaftCluster) handleDamagedStore(stats *pdpb.StoreStats) {
+	// ToDO: regions have no special process for the time being
+	// and need to be removed in the future
+	damaged_regions := stats.GetDamagedRegionsId()
+	if len(damaged_regions) == 0 {
+		return
+	}
+
+	log.Warn("store has damaged regions",
+		zap.Uint64("store-id", stats.GetStoreId()),
+		zap.Uint64s("region-ids", damaged_regions))
+	if exist, err := c.IsSchedulerExisted(schedulers.EvictLeaderName); !exist {
```

Comment:
Also, `c.IsSchedulerExisted` will acquire a read lock, if you put `handleDamagedStore` inside a lock itself, there will be a deadlock here.

Reply:
yep, and i found `c.GetOperatorController` and `c.AddScheduler` also acquire a lock.Looks like just unlock before `c.handleDamagedStore(stats)`?